File: CSharp-Part-2/6. Strings and Text Processing/StringsAndTextProcessing/ExtractHTML/ExtractHTML.cs
// Write a program that extracts from given HTML file its title (if available), and its body text without the HTML tags.

using System;
using System.Text.RegularExpressions;

class ExtractHTML
{
    static void Main()
    {
        string htmlStr = @"<html><head><title>News</title></head><body><p><a href=""http://academy.telerik.com"">TelerikAcademy</a> aims to provide free real-world practical training for young people who want to turn into skilful .NET software engineers.</p></body></html>";

        string title = Regex.Match(htmlStr, @"<title>.*?</title>").ToString();
        title = Regex.Replace(title, @"<.*?>", String.Empty);
        Console.Write("Title: ");
        Console.WriteLine(title);

        string text = Regex.Match(htmlStr, @"<body>.*?</body>").ToString();
        Console.Write("Text: ");
        Console.WriteLine(Regex.Replace(text, @"<.*?>", String.Empty));
    }
}

File: JavaScript-Fundamentals/11. Strings/extract-html.js
/*
	Write a function that extracts the content of a html page given as text.
	The function should return anything that is in a tag, without the tags.
*/

var contents = "<html><head><title>Sample site</title></head><body><div>text<div>more text</div>and more...</div>in body</body></html>";

function extractHTML(text) {
	var result = '',
		i;

	for (i = 0, len = text.length; i < len; i += 1) {
		if (text[i] === '<') {
			i = text.indexOf('>', i);
		} else {
			result += text[i];
		}
	}

	return result;
}

console.log(extractHTML(contents));